### ISSUE TITLE
mingw error for libuv v1.4.0: __declspec(align(8))

### ISSUE THREAD

**Issue by strrchr** [libuv/libuv#190]
libuv/src/win/fs.c:806:3: error: 'align' attribute directive ignored [-Werror=attributes]
   __declspec(align(8)) char buffer[8192];

msvc need: __declspec(align(8))
mingw need: __declspec(aligned(8))

**Comment by saghul** [comment]
What MinGW version are you using? I just tried cross-compiling with MinGW-w64 and got no problems.

**Comment by strrchr** [comment]
gcc version 4.9.2 (tdm64-1)

I also tested it using gcc version 4.9.2 (i686-win32-sjlj-rev0, Built by MinGW-W64 project).
I think GCC does not support 'align' in __declspec, so '__declspec(align(8))' in libuv/src/win/fs.c:806:3 compiled by GCC failed.

**Comment by Pagghiu** [comment]
I confirm this warning popping up on mingw 4.9.1 (the one that comes bundled with Qt Creator on Windows for example). It's a warning but in unfortunate case of buffer not aligned to 8 it will make the function fail at runtime.

The fix could be as simple as:

```C
#if _MSC_VER
  __declspec(align(8)) char buffer[8192];
#else
  __attribute__ ((aligned (8))) char buffer[8192];
#endif
```

**Comment by saghul** [comment]
@Pagghiu not sure why I wasn't seeing this before, but I tried your fix and it does indeed solve the warning. Do you mind submitting a PR?

**Comment by saghul** [comment]
Fixed in 70bbfa0, closing.
